portfolio/src/main/webapp/script.js

```diff
@@ -48,7 +48,9 @@ function createImg(imageName) {
  * Fetches the new comment and builds the UI.
  */
 function getComments() {
-  fetch('/data').then(response => response.json()).then((comments) => {
+  var maxComments = document.getElementById("commentsNum").value;
```

Comment:
Also here, prefer to keep consistency with the naming, so it'd be clear to the reader that this parameter addresses the same thing.